### PR TITLE
Microserver one step closer to being sinatra

### PR DESCRIPTION
**Pull request by kirillseva**
Add ability to serve static files through microserver.
This PR adds support for html, in a pretty crufty way. Need to rethink how this could be done prettier, and add support for js and images

### CODE REVIEW

File: R/routes.r [hunk 1]
```diff
@@ -1,5 +1,5 @@
 #' Determine the correct route to use based on a list of routes and
-#' the current path.
```
Comment (kirillseva) [inline R/routes.r]: Apparently path is a special word somewhere deep in R, and renaming the variable fixed my issues

File: R/routes.r [hunk 2]
```diff
@@ -7,15 +7,20 @@
...
-determine_route <- function(routes, path) {
+determine_route <- function(routes, request_path) {
   stopifnot(is.list(routes))
+  serve_static <- routes$serve_static %||% FALSE
   has_unnamed_route <-
     (identical(names(routes), NULL) && length(routes) != 0) || '' %in% names(routes)
   routenames <- names(routes)[names(routes) != '']
   for (route in routenames) {
-    if (grepl(paste0('^', route), path)) return(routes[[route]])
+    if (grepl(paste0('^', route), request_path)) {
```
Comment (robertzk) [inline R/routes.r]: Nice! Was going to add this.

**Comment by robertzk** [comment]
Can you backmerge master? I can't merge it.

File: R/microserver.r [hunk 3]
```diff
@@ -13,9 +13,8 @@ http_server <- function(routes) {
     params <- extract_params_from_request(req)
     query  <- extract_query_from_request(req)
     route  <- determine_route(routes, req$PATH_INFO)
-    # Provide access to the server environment.
-    environment(route) <-
-      list2env(list(.server_env = environment()), parent = environment(route))
+    if (is.microserver_response(route)) return(unclass(route))
+    environment(route) <- environment()
```
Comment (robertzk) [inline R/microserver.r]: I think we should keep the old one to remain compatible with the existing microservers. Otherwise route closures will break.

Comment (robertzk) [inline R/microserver.r]: i.e.,

```r
environment(route) <- list2env(list(.server_env = environment()), parent = environment(route))
```

**Pull request merged**